src/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2403,6 +2403,31 @@ private RuntimeTypeCache InitializeCache()
             return Cache.GetDefaultMemberName();
         }
 
+        internal IRuntimeMethodInfo? GetDefaultConstructor()
+        {
+            const BindingFlags bindingFlags = BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic;
+
+            RuntimeConstructorInfo[] cache = Cache.GetConstructorList(MemberListType.All, null);
```

Comment:
The custom attribute logic seems to be carefully written to avoid hitting reflection method caches that are expensive to populate. Can it be avoided?

The default constructor is cached on the VM side already. We should be able to take advantage of it.

Reply:
Pre-allocate the return value? e.g. cache the `IRuntimeMethodInfo` on the vm side?

Reply:
I would not materialize `IRuntimeMethodInfo` for the default constructor at all on this path. Would it be possible to modify the methods that consume this `IRuntimeMethodInfo` to treat null as "use default constructor if available"?